### PR TITLE
Aka task get

### PR DESCRIPTION
**Pull request by Alihan26**
# Pull Requests Checklist

Assignee:

- \[x\] code commented?
- \[x\] tests added / passed?
- \[x\] are tests covering all cases?
- \[x\] documentation added / updated?
- \[x\] pre-commit checks passed?
- \[x\] pipelines passed?

Reviewer:

- \[x\] are there any code smells? -> optional improvements, but otherwise ok
- \[x\] did pipeline pass? -> same as usual
- \[x\] is the code well documented? -> yes on confluence
- \[x\] are the tests passing and covering all cases?

### CODE REVIEW

File: src/main/java/ch/uzh/ifi/hase/soprafs24/controller/TeamController.java [hunk 1]
```diff
@@ -101,4 +101,20 @@ public TaskGetDTO createTask(@PathVariable("ID") Long teamId,
     // convert internal representation of task back to API
     return DTOMapper.INSTANCE.convertEntityToTaskGetDTO(createdTask);
   }
+
+  @GetMapping("/teams/{ID}/tasks")
+  @ResponseBody
+  public List<TaskGetDTO> getTasks(
+      @PathVariable("ID") Long teamId, @RequestHeader("Authorization") String token) {
+    // check if user is authorized (valid token)
+    authorizationService.isAuthorized(token);
```
Comment (5v3nn) [inline src/main/java/ch/uzh/ifi/hase/soprafs24/controller/TeamController.java:110]: Optional Improvement: use new `isAuthorizedAndInTeam` auth function

File: src/test/java/ch/uzh/ifi/hase/soprafs24/service/TaskServiceIntegrationTest.java [hunk 2]
```diff
@@ -150,4 +153,56 @@ public void createTask_nullDescription_throwsException() {
...
+
+    // given an invalid team id
+    Long invalidTeamId = validTeam.getTeamId() + 1;
+
+    // when & then
+    ResponseStatusException exception = assertThrows(
+        ResponseStatusException.class, () -> taskService.getTasksByTeamId(invalidTeamId));
+    assertEquals(HttpStatus.NOT_FOUND, exception.getStatus());
+    assertTrue(exception.getReason().contains("Team not found")); // adjusted assertion
+  }
```
Comment (5v3nn) [inline src/test/java/ch/uzh/ifi/hase/soprafs24/service/TaskServiceIntegrationTest.java:207]: optional improvement: add test to verify emtpy list if no tesks in team

File: src/test/java/ch/uzh/ifi/hase/soprafs24/service/TaskServiceTest.java [hunk 3]
```diff
@@ -121,11 +120,63 @@ public void createTask_nonExistingTeam_throwsException() {
...
+  /**
+   * Test for getting all tasks of a team if team exists but has no tasks
+   */
+  @Test
+  public void getTasksByTeamId_validInputs_noTasks() {
+    // when -> try to find teamId in the teamService -> return dummy team
+    Mockito.when(teamService.getTeamByTeamId(Mockito.any())).thenReturn(testTeam);
+
+    // when -> try to find tasks by team in the taskRepository -> return empty list
+    Mockito.when(taskRepository.findByTeam(Mockito.any())).thenReturn(new ArrayList<>());
```
Comment (5v3nn) [inline src/test/java/ch/uzh/ifi/hase/soprafs24/service/TaskServiceTest.java:159]: optional improvement: `thenReturn(List.of.empyt())` or something like that is better practice

**Review by 5v3nn — APPROVED** [review]
ok, you can merge

File: src/main/java/ch/uzh/ifi/hase/soprafs24/controller/TeamController.java (hunk 1 above)
Comment (Alihan26) [reply src/main/java/ch/uzh/ifi/hase/soprafs24/controller/TeamController.java:110]: all right, I will look in next merge, thanks

**Pull request merged**